sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1543,4 +1543,27 @@ class AdaptiveQueryExecSuite
     assert(materializeLogs(0).startsWith("Materialize query stage BroadcastQueryStageExec"))
     assert(materializeLogs(1).startsWith("Materialize query stage ShuffleQueryStageExec"))
   }
+
+  test("SPARK-34899: Use origin plan if we can not coalesce shuffle partition") {
+    def check(ds: Dataset[Row], origin: ShuffleOrigin): Unit = {
+      ds.collect()
+      val plan = ds.queryExecution.executedPlan.asInstanceOf[AdaptiveSparkPlanExec].executedPlan
+      assert(collect(plan) {
+        case c: CustomShuffleReaderExec => c
+      }.isEmpty)
+      assert(collect(plan) {
+        case s: ShuffleExchangeExec if s.shuffleOrigin == origin && s.numPartitions == 3 => s
+      }.size == 1)
+      checkAnswer(ds, testData)
+    }
+
+    withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
+      SQLConf.COALESCE_PARTITIONS_ENABLED.key -> "true",
+      SQLConf.ADVISORY_PARTITION_SIZE_IN_BYTES.key -> "10",
+      SQLConf.COALESCE_PARTITIONS_MIN_PARTITION_NUM.key -> "1",
+      SQLConf.SHUFFLE_PARTITIONS.key -> "3") {
+      check(testData.repartition(), REPARTITION)
```

Comment:
Can we create a new dataframe for testing here? It's unclear how many partitions `testData` has and what's the size for each partition.

Reply:
It is better to add the partition size of each partition in the comment.

Reply:
Just a question, is there a way to estimate the partition size quickly or just the type of jvm size ? Let's say we have one rows with one int column, and the map and reduce partition number are 1, then after shuffle the partition size is always 4.

Reply:
The partition size is stored in `ShuffleQueryStageExec#mapStats#bytesByPartitionId`. You can traverse the `bytesByPartitionId `value of each `mapStats `in [validMetrics ](https://github.com/apache/spark/blob/1237124062a541082922c650d1f1766b691f47bb/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CoalesceShufflePartitions.scala#L59).

Reply:
I see it, but want to find a way can estimate the size manual without that code ..

Reply:
`CoalesceShufflePartitions` rule depend on the map output statistics to coalesce the shuffle partitions. I think this size is more meaningful here.

Reply:
Aggre that specify the partition size explicitly is better, added the partition size comment.